micro-rdk/src/lib.rs

```diff
@@ -212,6 +212,9 @@ mod tests {
     pub fn global_network_test_lock<'a>() -> MutexGuard<'a, ()> {
         static LOCK: OnceLock<Mutex<()>> = OnceLock::new();
         let lock = LOCK.get_or_init(|| Mutex::new(()));
-        lock.lock().unwrap()
+        lock.lock().unwrap_or_else(|lock_result| {
+            lock.clear_poison();
+            lock_result.into_inner()
```

Comment:
I believe there shouldn't be an issue with multiple threads since:
a. we still hold the [LockResult](https://doc.rust-lang.org/std/sync/type.LockResult.html) which contains the guard
b. we return the guard

Reply:
Another reason why this is OK is that lock poisoning is mostly about ensuring that the state guarded by the mutex can be trusted to be in a consistent state. In this case, the state is unit, so there is no possibility of it having an inconsistent state.

Honestly this would probably be better written with a binary semaphore, since there isn't really state to be protected, but there doesn't seem to be such a thing in the standard library right now that we can use.

That said, I think there should be a comment here above `clear_poison` explaining why we feel justified to clear the poison state.